From 8eda79c0caf119ca0cbcf8b020fad80e09d8f5e6 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Fri, 12 Sep 2014 18:04:04 -0400
Subject: [powerpc] perf: Drop MMCRA from thread_struct

Message-id: <1410545655-205645-16-git-send-email-dzickus@redhat.com>
Patchwork-id: 94223
O-Subject: [RHEL7 PATCH 015/626] powerpc/perf: Drop MMCRA from thread_struct
Bugzilla: 1127366
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: David Gibson <dgibson@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1127366

commit 2ac138ca21ad26c988ce7c91d27327f85beb7519
Author: Michael Ellerman <michael@ellerman.id.au>
Date:   Fri Jun 28 18:15:15 2013 +1000

    powerpc/perf: Drop MMCRA from thread_struct

    In commit 59affcd "Context switch more PMU related SPRs" I added more
    PMU SPRs to thread_struct, later modified in commit b11ae95. To add
    insult to injury it turns out we don't need to switch MMCRA as it's
    only user readable, and the value is recomputed by the PMU code.

    Signed-off-by: Michael Ellerman <michael@ellerman.id.au>
    Signed-off-by: Benjamin Herrenschmidt <benh@kernel.crashing.org>

kabi notes:  Left behind an unused element 'mmcra' to preserve the size
of thread_struct which is embedded in task_struct.  Should be harmless.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/powerpc/include/asm/processor.h b/arch/powerpc/include/asm/processor.h
index d1c7667..4b3bf46 100644
--- a/arch/powerpc/include/asm/processor.h
+++ b/arch/powerpc/include/asm/processor.h
@@ -301,7 +301,7 @@ struct thread_struct {
  unsigned long mmcr2;
  unsigned  mmcr0;
  unsigned  used_ebb;
- unsigned long mmcra;
+ unsigned long mmcra; /* retain for KABI compliance */
 #endif
 };
 
diff --git a/arch/powerpc/kernel/asm-offsets.c b/arch/powerpc/kernel/asm-offsets.c
index 9c3d04a..30836e0 100644
--- a/arch/powerpc/kernel/asm-offsets.c
+++ b/arch/powerpc/kernel/asm-offsets.c
@@ -134,7 +134,6 @@ int main(void)
  DEFINE(THREAD_SIER, offsetof(struct thread_struct, sier));
  DEFINE(THREAD_MMCR0, offsetof(struct thread_struct, mmcr0));
  DEFINE(THREAD_MMCR2, offsetof(struct thread_struct, mmcr2));
- DEFINE(THREAD_MMCRA, offsetof(struct thread_struct, mmcra));
 #endif
 #ifdef CONFIG_PPC_TRANSACTIONAL_MEM
  DEFINE(PACATMSCRATCH, offsetof(struct paca_struct, tm_scratch));
-- 
1.7.1